server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -421,16 +421,18 @@ private static void parseObject(final DocumentParserContext context, String curr
         Mapper objectMapper = context.getMapper(currentFieldName);
         if (objectMapper != null) {
             context.path().add(currentFieldName);
-            if (objectMapper instanceof ObjectMapper objMapper) {
-                if (objMapper.subobjects() == false) {
-                    context.path().setWithinLeafObject(true);
-                }
-            }
```

Comment:
I would expect this to break things, as it effectively ignores the subobjects setting at each object level (it can be set within any object)

Reply:
After further investigation it indeed breaks things. It added a bug that is making the test `testSubobjectsFalseDocsWithInnerObjectMappedAsNonObject ` from the comment quoted below to pass because during the parsing of the document it retains `time` instead of `service.time` as the field name, therefore failing on retrieving the correct mapper. I'm adding the check back and I'll work on making the test pass!
> Does this test run? I remember the problem here being that we would provide the inner object to be parsed by the long `service.time` field, which would not work. That is because service.time is mapped, but not as an object. I am not completely following how you worked around that, I am also quite fuzzy on this because it's been a while since I worked on this issue.